cli/command/system/prune.go

```diff
@@ -0,0 +1,144 @@
+package system
+
+import (
+	"fmt"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/cli"
+	"github.com/docker/docker/cli/command"
+	units "github.com/docker/go-units"
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+)
+
+type pruneOptions struct {
+	force      bool
+	aggressive bool
+}
+
+// NewPruneCommand creates a new cobra.Command for `docker du`
+func NewPruneCommand(dockerCli *command.DockerCli) *cobra.Command {
+	var opts pruneOptions
+
+	cmd := &cobra.Command{
+		Use:   "prune [COMMAND]",
+		Short: "Delete unused data",
+		Args:  cli.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runPrune(dockerCli, true, true, true, opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVarP(&opts.force, "force", "f", false, "Do not prompt for confirmation")
+	flags.BoolVarP(&opts.aggressive, "aggressive", "a", false, "Delete all unused images not just dangling ones")
+
+	cmd.AddCommand(
+		&cobra.Command{
+			Use:   "containers",
+			Short: "Delete unused containers",
+			Args:  cli.NoArgs,
+			RunE: func(cmd *cobra.Command, args []string) error {
+				return runPrune(dockerCli, true, false, false, opts)
+			},
+		},
+		&cobra.Command{
+			Use:   "volumes",
+			Short: "Delete unused volumes",
+			Args:  cli.NoArgs,
+			RunE: func(cmd *cobra.Command, args []string) error {
+				return runPrune(dockerCli, false, true, false, opts)
+			},
+		},
+		&cobra.Command{
+			Use:   "images",
+			Short: "Delete unused images",
+			Args:  cli.NoArgs,
+			RunE: func(cmd *cobra.Command, args []string) error {
+				return runPrune(dockerCli, false, false, true, opts)
+			},
+		},
+	)
+
+	return cmd
+}
+
+func runPrune(dockerCli *command.DockerCli, deleteContainers, deleteVolumes, deleteImages bool, opts pruneOptions) error {
+	if opts.force == false {
+		fmt.Fprintln(dockerCli.Out(), "WARNING! This will remove:")
+		if deleteContainers {
+			fmt.Fprintln(dockerCli.Out(), "\t- all stopped containers")
+		}
+		if deleteVolumes {
+			fmt.Fprintln(dockerCli.Out(), "\t- all volumes not used by at least one container")
+		}
+		if deleteImages {
+			fmt.Fprintln(dockerCli.Out(), "\t- all images without at least one container associated to them")
+		}
+		fmt.Fprintf(dockerCli.Out(), "Are you sure you want to continue? [y/N] ")
+
+		answer := ""
+		n, _ := fmt.Fscan(dockerCli.In(), &answer)
+		if n != 1 || (answer != "y" && answer != "Y") {
+			return nil
+		}
+	}
+
+	var spaceReclaimed uint64
+
+	if deleteContainers {
+		report, err := dockerCli.Client().ContainersPrune(context.Background(), types.ContainersPruneConfig{})
+		if err != nil {
+			return err
+		}
+		if len(report.ContainersDeleted) > 0 {
+			fmt.Fprintln(dockerCli.Out(), "Deleted Containers:")
+			for _, id := range report.ContainersDeleted {
+				fmt.Fprintln(dockerCli.Out(), id)
+
+			}
+			fmt.Fprintln(dockerCli.Out(), "")
+			spaceReclaimed += report.SpaceReclaimed
+		}
+	}
```

Comment:
If we're going to keep both, this should probably just call the `runPrune` from `cli/command/container/prune.go`